api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -51,6 +54,10 @@ message UdpProxyConfig {
 
     // The upstream cluster to connect to.
     string cluster = 2 [(validate.rules).string = {min_len: 1}];
```

Comment:
I still think we should deprecate this, and I would like to close out the comment thread and how to have a trivial generic matching config that would just always route to a single cluster, and use that in all of the examples. @snowp can you help with this thread of this PR? Thank you.

Reply:
https://github.com/envoyproxy/envoy/blob/0458877b46f9bcb96f843331e2515f7aaf6bb800/api/envoy/config/common/matcher/v3/matcher.proto#L130-L144
I have a suggestion, shall we remove the requirement of this `matcher_type`, and change the name of `on_no_match` to `default_match`. Then we can have an only `default_match` for routing to a single cluster.

Reply:
I think perhaps removing the requirement of a match would work yeah. We can't just rename fields, but we could potentially go through a deprecation cycle if we wanted (cc @envoyproxy/api-shepherds)

Also I just realized we want to be using the https://github.com/cncf/xds/blob/main/xds/type/matcher/v3/matcher.proto proto here, not the old config/common/matcher one

Reply:
Alternatively we could extend SinglePredicate to have an always true branch:

```
      oneof match_type {
        option (validate.required) = true;

        // A single predicate to evaluate.
        SinglePredicate single_predicate = 1;

        // A list of predicates to be OR-ed together.
        PredicateList or_matcher = 2;

        // A list of predicates to be AND-ed together.
        PredicateList and_matcher = 3;

        // The invert of a predicate
        Predicate not_matcher = 4;
      }
```

e.g. a `bool match_all` oneof option. Perhaps a little less elegant but it would also do the trick

Reply:
So, I think we shall switch to xds/matcher, and use `matcher_list` instead of `matcher_tree` in the example to make it simpler. And we can remove the requirement in the future.

Reply:
Ok, I have tested with `matcher_list`, but I think it is much more verbose then `matcher_tree`

```yaml
matcher_list:
  matchers:
  - predicate:
      single_predicate:
        input:
          name: source-ip
          typed_config:
            '@type': type.googleapis.com/envoy.type.matcher.v3.SourceIpMatchInput
        value_match:
          exact: "127.0.0.1"
    on_match:
      action:
        name: route
        typed_config:
          '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.Route
          cluster: service_udp

matcher_tree:
  input:
    name: source-ip
    typed_config:
      '@type': type.googleapis.com/envoy.type.matcher.v3.SourceIpMatchInput
  exact_match_map:
    map:
      "127.0.0.1":
        action:
          name: route
          typed_config:
            '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.Route
            cluster: service_udp
```

Reply:
I have re-deprecated the original `cluster`. So shall we use the `matcher_tree` or wait the xDS API update for a concise syntax like remove the requirement of `matcher_tree` and `matcher_list`.

Reply:
I think both of these are still verbose if we just want to universally always route to a cluster without any matching effectively. @snowp what do you recommend here? Can we just make the matcher change now to support something less verbose for a universal match?

/wait

Reply:
Started a PR here https://github.com/cncf/xds/pull/19 ptal

Reply:
Since related PRs have been merged, we can now write
```
on_no_match:
  action:
    name: route
    typed_config:
      '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.Route
      cluster: service_udp
```
to serve a single UDP cluster. And it is much simpler than the previous one.